compiler/crates/common/src/text_source.rs

```diff
@@ -113,6 +113,13 @@ impl TextSource {
                 character += 1;
             }
         }
+
+        if start_position != lsp_types::Position::default()
+            && end_position == lsp_types::Position::default()
+        {
+            end_position = lsp_types::Position::new(line as u32, character as u32);
+        }
```

Comment:
There was a bug here in that if the diagnostic pointed to the last character you would always get a span where the end would point to 0,0. Now if you have a start, but no end, it uses the last character as the span's end.